File: uikit/src/main/java/viewmodel/SmartReplyViewModel.java
package viewmodel;

import android.content.Context;

import com.cometchat.pro.uikit.SmartReplyList;

import java.util.List;

import adapter.SmartReplyListAdapter;
import listeners.OnItemClickListener;

public class SmartReplyViewModel {

    private static final String TAG = "SmartReplyViewModel";

    private Context context;

    private SmartReplyListAdapter smartReplyListAdapter;

    private SmartReplyList smartReplyList;

    public SmartReplyViewModel(Context context, SmartReplyList smartReplyList) {
        this.context = context;
        this.smartReplyList = smartReplyList;
         setSmartReplyAdapter(smartReplyList);
    }

    private void setSmartReplyAdapter(SmartReplyList smartReplyList) {
        smartReplyListAdapter=new SmartReplyListAdapter(context);
        smartReplyList.setAdapter(smartReplyListAdapter);
    }

    private SmartReplyListAdapter getAdapter(){
        if (smartReplyListAdapter==null){
            smartReplyListAdapter=new SmartReplyListAdapter(context);
        }
        return smartReplyListAdapter;
    }

    public void setSmartReplyList(List<String> replyList){
        getAdapter().updateList(replyList);
    }


}

File: uikit-kotlin/src/main/java/viewmodel/SmartReplyViewModel.kt
package viewmodel

import adapter.SmartReplyListAdapter
import android.content.Context
import com.cometchat.pro.uikit.SmartReplyList

class SmartReplyViewModel(private val context: Context, private val smartReplyList: SmartReplyList) {
    private var smartReplyListAdapter: SmartReplyListAdapter? = null
    private fun setSmartReplyAdapter(smartReplyList: SmartReplyList) {
        smartReplyListAdapter = SmartReplyListAdapter(context)
        smartReplyList.adapter = smartReplyListAdapter
    }

    private val adapter: SmartReplyListAdapter
        private get() {
            if (smartReplyListAdapter == null) {
                smartReplyListAdapter = SmartReplyListAdapter(context)
            }
            return smartReplyListAdapter!!
        }

    fun setSmartReplyList(replyList: List<String>?) {
        adapter.updateList(replyList!!)
    }

    companion object {
        private const val TAG = "SmartReplyViewModel"
    }

    init {
        setSmartReplyAdapter(smartReplyList)
    }
}